src/extensions/default/InlinePaddingEditor/PaddingEditor.js

```diff
@@ -0,0 +1,212 @@
+            return $(this).text().trim() === self.unit;
+        }).addClass("selected");
+        $text.text(self.toString());
+
+        $slider.on("input", function() {
+            var newValue = $slider.val().trim();
+            self.value = newValue;
+            $text.text(self.toString());
+            onChange();
+        });
+
+        $unitOptions.on("click", function() {
+            var $selectedUnit = $(this);
+            $selectedUnit.siblings().removeClass("selected");
+            $selectedUnit.addClass("selected");
+
+            self.unit = $selectedUnit.text().trim();
+            $text.text(self.toString());
+            onChange();
+        });
+    }
+
+    PaddingValue.prototype.toString = function() {
+        return this.value + (this.value === 0 ? "" : this.unit);
+    };
+
+    function PaddingEditor($parent, valueString, paddingChangeHandler) {
+        var self = this;
+
+        // Create the DOM structure, filling in localized strings via Mustache
+        self.$element = $(Mustache.render(PaddingTemplate, Strings));
+        $parent.append(self.$element);
+        self.paddingChangeHandler = paddingChangeHandler;
+
+        this.onChange = this.onChange.bind(this);
+        self.updateValues(valueString);
+
+        // Attach event listeners to toggle the corner mode UI elements
+        var $individualSidesArea = self.$element.find("#individualSidesArea");
+        var $individualSidesButton = self.$element.find("#individualSides");
+        var $allSidesArea = self.$element.find("#allSidesArea");
+        var $allSidesButton = self.$element.find("#allSides");
+
+        function toggleSidesOption($showElement, $hideElement) {
+            $showElement.show();
+            $hideElement.hide();
+            self.allSides = $showElement === $allSidesArea;
+            check = $showElement === $allSidesArea;
+            self.onChange();
+        }
+
+        $allSidesButton.on("click", function() {
+            $allSidesButton.addClass("selected");
+            $individualSidesButton.removeClass("selected");
+            toggleSidesOption($allSidesArea, $individualSidesArea);
+        });
+        $individualSidesButton.on("click", function() {
+            $allSidesButton.removeClass("selected");
+            $individualSidesButton.addClass("selected");
+            toggleSidesOption($individualSidesArea, $allSidesArea);
+        });
+
+        // initialize individual side editing to be disabled if allSides is set to true
+        if(self.allSides){
+            $allSidesButton.trigger("click");
+        } else {
+            $individualSidesButton.trigger("click");
+        }
+    }
+
+    PaddingEditor.prototype.updateValues = function(valueString) {
+        var values = getIndividualValues(valueString);
+        var numOfValues = values.length;
+        var firstValue = values[0];
+        var secondValue = firstValue;
+        var thirdValue = firstValue;
+        var fourthValue = firstValue;
```

Comment:
Can you leave a comment to explain why you are using `firstValue` in all cases?  I'm sure it's fine, but it's not obvious (to me).